File: IT/1/Oggetti/Esercizi/Java/Esercizio28.java
/* file Esercizio28.java */

/* la classe Circle rappresenta una circonferenza */

class Circle {
	/* ogni circonferenza viene definita dal suo raggio */
	private double r;

	public Circle(double r) {
		this.r = r;
	}

	/* area del cerchio = Pi greco * raggio al quadrato */
	public double area() {
		return Math.PI * this.r * this.r;
	}
}

/* la classe Cone rappresenta un cono tridimensionale */

class Cone {
	private Circle base;
	private double h;

	/* ogni cono viene definito da altezza e circonferenza alla base */
	public Cone(Circle base, double h) {
		this.base = base;
		this.h = h;
	}

	/* volume del cono: (3.14 * r^2 * h) / 3 */
	public double volume() {
		return (this.base.area() * this.h) / 3;
	}
}

public class Esercizio28 {
	public static void main(String [] args) {
		Circle base = new Circle(2.0);
		Cone solid = new Cone(base, 3.0);

		System.out.println("Area circonferenza alla base: " + String.format("%.2f", base.area()) + " centimetri quadrati.");
		System.out.println("Volume cono: " + String.format("%.2f", solid.volume()) + " centimetri cubi.");
	}
}

File: IT/1/Oggetti/Esercizi/Python/esercizio28.py
# file esercizio28.py

from __future__ import print_function
from math import pi

# la classe Circle rappresenta una circonferenza

class Circle(object):
	# ogni circonferenza viene definita dal suo raggio
	def __init__(self, r):
		self.__r = r

	# area del cerchio = Pi greco * raggio al quadrato
	def area(self):
		return pi * self.__r * self.__r

# la classe Cone rappresenta un cono tridimensionale

class Cone(object):
	# ogni cono viene definito da altezza e circonferenza alla base
	def __init__(self, base, h):
		self.__base = base
		self.__h = h

	# volume del cono: (3.14 * r^2 * h) / 3
	def volume(self):
		return (self.__base.area() * self.__h) / 3

circle = Circle(r=2.0)

print("Area circonferenza alla base: {:.2f} centimetri quadrati.".format(circle.area()))

solid = Cone(base=circle, h=3.0)

print("Volume cono: {:.2f} centimetri cubi.".format(solid.volume())) 

